server/app/services/applications/ProgramAdminApplicationService.java

```diff
@@ -58,4 +64,20 @@ public void setStatus(Application application, StatusEvent newStatus, Account ad
             application, admin, ApplicationEventDetails.Type.STATUS_CHANGE, details);
     eventRepository.insertSync(event);
   }
+
+  /**
+   * Saves a new {@link ApplicationEventDetails.Type.NOTE_CHANGE} event.
+   *
+   * @param admin The Account that instigated the change.
+   */
+  public void setNote(Application application, NoteEvent note, Account admin) {
+    ApplicationEventDetails details =
+        ApplicationEventDetails.builder()
+            .setEventType(ApplicationEventDetails.Type.NOTE_CHANGE)
+            .setNoteEvent(note)
+            .build();
+    ApplicationEvent event =
+        new ApplicationEvent(application, admin, ApplicationEventDetails.Type.NOTE_CHANGE, details);
```

Comment:
Didn't notice originally, but why do we have to specify the ApplicationEventDetails.Type both in the ApplicationEventDetails builder as well as the ApplicationEvent constructor? Could we remove the argument from the constructor and instead infer it from the provided details? That would cut out a class of potential mismatch errors that might come our way and simplify usage.

Reply:
They both have different utility, one is at the database level to query on, the other is that I think it's reasonable to want to pass the Details object rather than the entire Event.  I might be wrong there, but felt better to start with than add subsequently.

Reply:
Put another way, is there ever a case where we expect the two to mismatch? If not, then I don't see the need for it in the constructor. EBean's population of an ApplicationEvent won't use the constructor, I don't believe. It will instead use the provided getters/setters. Constructor initialization is only for our application code, where I don't think there's a need to specify them separately.

Reply:
Ah sorry, I missed the second infer part of your original comment, remove the param in the ctor.